Undefined country code when selecting out of range address

MJQ7 opened the pull request:
https://eaflood.atlassian.net/browse/LTFRI-2246

This ticket will fix an uncommon bug where session data gets mixed up if the user has the service open in multiple tabs

MJQ7 committed:
Undefined country code when selecting out of range address

https://eaflood.atlassian.net/browse/LTFRI-2246

This ticket will fix an uncommon bug where session data gets mixed up if the user has the service open in multiple tabs

MJQ7 committed:
Enhanced postcode validation and error handling in search functionality.

ADDED: postcode mismatch check for address selection
ADDED: out of bounds check for address index
UPDATED: search route to redirect on postcode mismatch

Changes:
	search.js updated
	search.spec.js updated

Matthew-Collins-Defra commented on server/routes/search.js:
```diff
@@ -98,9 +98,15 @@ module.exports = [
       const addresses = request.yar.get('addresses')
 
+      // throw for postcode mismatch when address is within addresses index range
+      if (addresses?.length > 0 && postcode !== addresses?.[0]?.postcode) {
```
This is causing the search to fail if the postcode doesn't match exactly. For example. I type 'wa41ab' into the postcode search, it doesn't match 'WA4 1AB' so the search always goes back to the postcode entry screen.

MJQ7 committed:
Added postcode normalisation in search functionality.

ADDED: `normalisePostcode` function
UPDATED: search route to use normalisePostcode
ADDED: tests for normalisePostcode

Changes:
    helpers.js updated
    search.js updated
    helpers.spec.js created

Matthew-Collins-Defra reviewed (approved):
lgtm!
